spring-batch-core/src/main/java/org/springframework/batch/core/configuration/annotation/EnableBatchProcessing.java

```diff
@@ -85,11 +85,8 @@
  * }
  * </pre>
  *
- * If multiple {@link javax.sql.DataSource}s are defined in the context, the one annotated with
- * {@link org.springframework.context.annotation.Primary} will be used (Note that if none
- * of them is annotated with {@link org.springframework.context.annotation.Primary}, the one
- * named <code>dataSource</code> will be used if any, otherwise a {@link UnsatisfiedDependencyException}
- * will be thrown).
+ * If multiple {@link javax.sql.DataSource}s are defined in the context, the primary autowire candidate
```

Comment:
No, we are not losing behavior, we are only requiring at least one datasource (because as mentioned above, it is not possible to create a default `BatchConfigurer` without a datasource anymore). The error message was incorrect indeed, I fixed it accordingly (changed the wording from `precisely one DataSource` to `at least one datasource` (see da506ef03a810e9a6d7a2f1e8376170ca30a4dfb). I also updated the exception handling to be more specific for each case. Thank you for raising this! The updated version is now consistent with the Javadoc.

Reply:
Sweet!   LGTM